httpx/_decoders.py

```diff
@@ -267,57 +267,23 @@ def __init__(self) -> None:
         self.buffer = ""
 
     def decode(self, text: str) -> typing.List[str]:
-        lines = []
-
-        if text and self.buffer and self.buffer[-1] == "\r":
-            if text.startswith("\n"):
-                # Handle the case where we have an "\r\n" split across
-                # our previous input, and our new chunk.
-                lines.append(self.buffer[:-1] + "\n")
-                self.buffer = ""
-                text = text[1:]
-            else:
-                # Handle the case where we have "\r" at the end of our
-                # previous input.
-                lines.append(self.buffer[:-1] + "\n")
-                self.buffer = ""
-
-        while text:
-            num_chars = len(text)
-            for idx in range(num_chars):
-                char = text[idx]
-                next_char = None if idx + 1 == num_chars else text[idx + 1]
-                if char == "\n":
-                    lines.append(self.buffer + text[: idx + 1])
-                    self.buffer = ""
-                    text = text[idx + 1 :]
-                    break
-                elif char == "\r" and next_char == "\n":
-                    lines.append(self.buffer + text[:idx] + "\n")
-                    self.buffer = ""
-                    text = text[idx + 2 :]
-                    break
-                elif char == "\r" and next_char is not None:
-                    lines.append(self.buffer + text[:idx] + "\n")
-                    self.buffer = ""
-                    text = text[idx + 1 :]
-                    break
-                elif next_char is None:
-                    self.buffer += text
-                    text = ""
-                    break
+        if self.buffer:
+            text = self.buffer + text
+
+        if text.endswith("\r"):
+            self.buffer = text
+            return []
+
+        lines = text.splitlines()
+        if text.endswith("\n") or not lines:
```

Comment:
The `if text.endswith("\n")` doesn't seems sufficient to me.

Other newline endings [are also available](https://docs.python.org/3/library/stdtypes.html#str.splitlines).

I'd assume that our test cases aren't covering this sufficiently.

Reply:
Yeah rn they only care about CR, LF, and CRLF

Reply:
The behaviour when using any of the other "funny money" line-endings is not really incorrect output, but that trailing lines will be delayed until the subsequent call or until finalization when one or two lines will be returned.

Reply:
Well, unless you consider splitting on those other line endings is, itself, as a bug/unintended. (I had no idea, fwiw)